nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/EncryptAttributes.java

```diff
@@ -0,0 +1,611 @@
+
+    private static AllowableValue[] buildEncryptionMethodAllowableValues() {
+        final EncryptionMethod[] encryptionMethods = EncryptionMethod.values();
+        List<AllowableValue> allowableValues = new ArrayList<>(encryptionMethods.length);
+        for (EncryptionMethod em : encryptionMethods) {
+            allowableValues.add(new AllowableValue(em.name(), em.name(), em.toString()));
+        }
+
+        return allowableValues.toArray(new AllowableValue[0]);
+    }
+
+    private static AllowableValue[] buildWeakCryptoAllowableValues() {
+        List<AllowableValue> allowableValues = new ArrayList<>();
+        allowableValues.add(new AllowableValue(WEAK_CRYPTO_ALLOWED_NAME, "Allowed", "Operation will not be blocked and no alerts will be presented " +
+                "when unsafe combinations of encryption algorithms and passwords are provided"));
+        allowableValues.add(buildDefaultWeakCryptoAllowableValue());
+        return allowableValues.toArray(new AllowableValue[0]);
+    }
+
+    private static AllowableValue buildDefaultWeakCryptoAllowableValue() {
+        return new AllowableValue(WEAK_CRYPTO_NOT_ALLOWED_NAME, "Not Allowed", "When set, operation will be blocked and alerts will be presented to the user " +
+                "if unsafe combinations of encryption algorithms and passwords are provided on a JVM with limited strength crypto. To fix this, see the Admin Guide.");
+    }
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        final List<PropertyDescriptor> properties = new ArrayList<>();
+        properties.add(ATTRIBUTES_TO_ENCRYPT);
+        properties.add(MODE);
+        properties.add(KEY_DERIVATION_FUNCTION);
+        properties.add(ENCRYPTION_ALGORITHM);
+        properties.add(ALLOW_WEAK_CRYPTO);
+        properties.add(PASSWORD);
+        properties.add(RAW_KEY_HEX);
+        properties.add(PUBLIC_KEYRING);
+        properties.add(PUBLIC_KEY_USERID);
+        properties.add(PRIVATE_KEYRING);
+        properties.add(PRIVATE_KEYRING_PASSPHRASE);
+        this.properties = Collections.unmodifiableList(properties);
+
+        final Set<Relationship> relationships = new HashSet<>();
+        relationships.add(REL_SUCCESS);
+        relationships.add(REL_FAILURE);
+        this.relationships = Collections.unmodifiableSet(relationships);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return properties;
+    }
+
+    public static boolean isPGPAlgorithm(final String algorithm) {
+        return algorithm.startsWith("PGP");
+    }
+
+    public static boolean isPGPArmoredAlgorithm(final String algorithm) {
+        return isPGPAlgorithm(algorithm) && algorithm.endsWith("ASCII-ARMOR");
+    }
+
+    @Override
+    protected Collection<ValidationResult> customValidate(final ValidationContext context) {
```

Comment:
Much of this validation code is now duplicated from `EncryptContent` and should be consolidated, as future enhancements to keyed encryption will modify this code in multiple places.